zod/cli/torrent_download.py

```diff
@@ -0,0 +1,347 @@
+            files.append("images_front_blur")
+        # we only have dnat images for frames
+        if filter_settings.dnat and subset == SubDataset.FRAMES:
+            files.append("images_front_dnat")
+
+    # gnss data
+    if filter_settings.oxts:
+        files.append("oxts")
+
+    # lidar_velodyne data
+    if filter_settings.lidar:
+        files.append("lidar_velodyne")
+        if subset == SubDataset.FRAMES:
+            if not filter_settings.num_scans_before == 0:
+                n_scans = (
+                    10
+                    if filter_settings.num_scans_before == -1
+                    else filter_settings.num_scans_before
+                )
+                for i in range(1, n_scans + 1):
+                    files.append(f"lidar_velodyne_{i:02d}before")
+
+            if not filter_settings.num_scans_after == 0:
+                n_scans = (
+                    10 if filter_settings.num_scans_after == -1 else filter_settings.num_scans_after
+                )
+                for i in range(1, n_scans + 1):
+                    files.append(f"lidar_velodyne_{i:02d}after")
+
+    # vehicle data
+    if filter_settings.vehicle_data:
+        if subset != SubDataset.FRAMES:
+            files.append("vehicle_data")
+
+    # frame infos
+    if filter_settings.infos:
+        files.append("infos")
+
+    file_indicies = [files_to_idx[f] for f in files]
+
+    files = [f"{file}.tar.gz" for file in files]
+
+    return files, file_indicies
+
+
+def download_torrent_file(subset: SubDataset, output_dir: Path):
+    """Download the torrent file for the given subset and version."""
+    url = TORRENT_FILEMAP[subset]
+    filename = f"{subset.name.lower()}.torrent"
+    print("Downloading torrent file: {}".format(filename))
+    try:
+        urllib.request.urlretrieve(url, filename)
+        # move the torrent file to the output directory
+        os.rename(filename, output_dir / filename)
+
+        assert os.path.getsize(output_dir / filename) == SIZE_FILEMAP[subset], (
+            f"Downloaded torrent file is not the correct size. "
+            f"Expected: {SIZE_FILEMAP[subset]}, "
+            f"Actual: {os.path.getsize(output_dir / filename)}"
+        )
+        return output_dir / filename
+    except Exception as e:
+        print(f"Failed to download torrent file: {e}")
+        exit(1)
+
+
+def _download_dataset(
+    dl_settings: DownloadSettings, filter_settings: FilterSettings, subset: SubDataset
+):
+    dl_dir = Path(dl_settings.output_dir) / "downloads"
+    dl_dir.mkdir(parents=True, exist_ok=True)
+
+    torrent_filepath = download_torrent_file(subset, dl_dir)
+
+    files, file_indicies = get_files_to_download(subset, filter_settings)
+    # create the aria2c command
+    cmd = [
+        "aria2c",
+        f"--select-file={','.join([str(i) for i in file_indicies])}",
+        f"--torrent-file={torrent_filepath}",
+        f"--dir={str(dl_dir)}",
+        "--seed-time=0",
+        "--continue=true",
```

Comment:
would be nice with a comment explaining what continue does (maybe seed time too to be sure)

Reply:
Added explanations.